modules/engine-prototype/src/main/scala/engine/Sequence.scala

```diff
@@ -0,0 +1,91 @@
+package engine
+
+import scala.concurrent.ExecutionContext
+
+import cats.{FlatMap, Functor}
+import cats.data.NonEmptyList
+import cats.effect.Effect
+import cats.effect.implicits._
+import cats.implicits._
+
+import fs2.Stream
+import fs2.async
+import fs2.async.mutable.Signal
+
+import monocle.{Lens, Prism}
+import monocle.macros.GenLens
+
+sealed trait Sequence
+object Sequence {
+
+  sealed trait F2 extends Sequence
+  object F2 {
+
+    final case class Ready(
+      done: List[Step.F2.Done],
+      pending: List[Step.F2.Pending],
+      current: Step.F2.Current
+    ) extends F2
+
+    final case class Done(done: NonEmptyList[Step.F2.Done]) extends F2
+
+  }
+
+  final case class State(sequence: Sequence, status: Status)
+
+  object State {
+
+    val sequence: Lens[State, Sequence] = GenLens[State](_.sequence)
+
+    val status: Lens[State, Status] = GenLens[State](_.status)
+
+    val current: Prism[State, Step.F2.Current] = ???
+
+    val standard: Prism[State, Step.F2.Current.Pending.Standard] = ???
+
+    val ready: Prism[State, Step.F2.Current.Pending] = ???
+
+    // Thread safe mutable but careful need to be careful not to block for very long.
+    case class Mutable[F[_]](signal: Signal[F, Sequence.State]) {
```

Comment:
The underlying concurrency is `fs2` `Signal` but in the rest of the code we shouldn't mess directly with `Signal`s,  instead we create the wrappers that match more closely the terminology we use in the Seqexec, simpler more specialized signatures and also by having all these State manipulation in an object with conspicuous name like `Mutable` we make sure you should watch out what you do when using this functions.

In reality the danger refers that we may block for too long (something that also have to be careful in our current seqexec), there are no fundamental dangerous operations with race conditions or deadlocks.